src/routes/mod_creation.rs

```diff
@@ -0,0 +1,401 @@
+                                    )
+                                    .await?;
+
+                                    created_version.files.push(VersionFile {
+                                        game_versions: version_data
+                                            .game_versions
+                                            .into_iter()
+                                            .map(|x| x.0)
+                                            .collect(),
+                                        hashes: vec![FileHash {
+                                            algorithm: "sha1".to_string(),
+                                            hash: upload_data.content_sha1,
+                                        }],
+                                        url: format!("{}/{}", cdn_url, upload_data.file_name),
+                                    });
+                                }
+                                None => {
+                                    let mut version_id = VersionId(random_base62(8));
+                                    //Check if ID is unique
+                                    loop {
+                                        let filter = doc! { "_id": version_id.0 };
+
+                                        if versions.find(filter, None).await?.next().await.is_some()
+                                        {
+                                            version_id = VersionId(random_base62(8));
+                                        } else {
+                                            break;
+                                        }
+                                    }
+
+                                    let body_url = format!(
+                                        "data/{}/changelogs/{}/body.md",
+                                        mod_id.0, version_id.0
+                                    );
+
+                                    upload_file(
+                                        upload_url.get_ref().clone(),
+                                        "text/plain".to_string(),
+                                        body_url.clone(),
+                                        version_data.version_body.into_bytes(),
+                                    )
+                                    .await?;
+
+                                    let upload_data = upload_file(
+                                        upload_url.get_ref().clone(),
+                                        "application/java-archive".to_string(),
+                                        format!(
+                                            "{}/{}/{}",
+                                            create_data.mod_namespace.replace(".", "/"),
+                                            version_data.version_slug,
+                                            file_name
+                                        ),
+                                        (&data).to_owned().to_vec(),
```

Comment:
This appears to be a double allocation.  The `upload_file` function should take a slice, rather than a `Vec<u8>`.

Reply:
Will fix this.

Reply:
Okay, I figured out why I didn't do this earlier, reqwest doesn't take in a slice for it's body for some reason. (they do but only one with a static lifetime)

Reply:
Hmm.  That's not optimal, but we should still be able to reduce this case to a single allocation rather than two.